### ISSUE TITLE
build on RHEL with dnf

### ISSUE THREAD

**Issue by antonio-fr** [pypa/setuptools#1026]
On latest RHEL system, 'rpm' is replaced by default with 'dnf'
When build scripts call 'gcc', this throws "no such file /usr/lib/rpm/redhat..."

It takes time for me to find out this issue. This can be fixed with
dnf install redhat-rpm-config

It would be good to upgrade script to take in account modern "dnf systems" so that it doesn't require additional 'rpm'.

**Comment by JonathonReinhart** [comment]
> On latest RHEL system, 'rpm' is replaced by default with 'dnf'

I don't think this is quite correct. `rpm` is the low-level mechanism which manages the installation of packages in RPM format. Previous Fedora / Red Hat based distros (e.g. Fedora, RHEL, CentOS) used **`yum`** as their package manager which handles retrieving RPMs from remote repositories and installing them (using `rpm`). Newer distros have replaced `yum` with `dnf`. AFAIK, `rpm` is *still* used under the hood to manage installing RPM packages.
